staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -527,6 +527,15 @@ func (cnc *CloudNodeController) getProviderID(ctx context.Context, node *v1.Node
 		return node.Spec.ProviderID, nil
 	}
 
+	if instancesV2, ok := cnc.cloud.InstancesV2(); ok {
+		metadata, err := instancesV2.InstanceMetadata(ctx, node)
+		if err != nil {
```

Comment:
I think we need to check for `cloudprovider.NotImplemented` error similar to line 540

Reply:
gotcha, will update!

Reply:
Actually, I think I change my mind. I'm not sure it's really helpful to handle NotImplemented error from `InstanceMetadata`

Reply:
removed, I think we can just return an error there